remove resource pages and child models

scytacki opened the pull request:
Resource pages are not really used by project portals. Interactions does have 14 resource pages, but they are used as a way to upload files so the files can be used by activities. The team now uses Cyberduck or some other tool to upload files directly to their S3 bucket.

This PR removes all of the resource page support in the portal. This consists of 3 models:
- ResourcePage
- StudentView
- AttachedFile
The two additional models are only used by resource pages so they can be deleted too.

Additionally I deleted some themes that aren't needed anymore

scytacki commented:
@knowuh or @pjanik would you mind skimming this?  
It is mostly just removing things.

dougmartin commented:
Looks good to me.  That is a lot of code removed!

knowuh commented on app/models/external_activity.rb:
```diff
@@ -121,12 +121,6 @@ def valid_url
 
   scope :ordered_by, lambda { |order| { :order => order } }
-  # FIXME: See comments in app/models/resource_page.rb
-  scope :match_any, lambda { |scopes|
```
To check:  Nothing uses ExternalActivity.match_any

knowuh replied:
Confirmed it myself:  Nothing uses `#match_any` scope.

knowuh commented on db/schema.rb:
```diff
@@ -2570,7 +2534,6 @@
 
   add_index "taggings", ["tag_id"], :name => "index_taggings_on_tag_id"
   add_index "taggings", ["taggable_id", "taggable_type", "context"], :name => "index_taggings_on_taggable_id_and_taggable_type_and_context"
-  add_index "taggings", ["tagger_id", "tagger_type"], :name => "index_taggings_on_tagger_id_and_tagger_type"
```
The removal of this index seems unrelated?

knowuh commented:
This looks good.  Two items I wasn't sure about. 

LOVE all the vanished code.  :+1:

scytacki commented:
Thanks for looking at it!
@knowuh I added back in those indexes, I'm not sure why they were removed.
Yes I removed `match_any` because I couldn't find it used anywhere either.

scytacki commented:
this was passing all of the tests before and the push is passing the tests. So I'm pretty sure the failures are random. I'm going to merge this and get it over with.
